Enable users to upload and view facility lists

kellyi opened the pull request:
## Overview

This PR enables users to upload and view facility lists through the UI.

Closes #101 
Connects #148
Connects #105 

## Demo

![facility-lists](https://user-images.githubusercontent.com/4165523/51869124-ab82a380-231d-11e9-878f-9a2f9f0adbed.gif)
![facility-lists-2](https://user-images.githubusercontent.com/4165523/51869125-ab82a380-231d-11e9-86a8-6e9ea0e69c11.gif)

## Testing Instructions

- get this branch then
- `./scripts/manage reset_schema`
- `./scripts/manage migrate`
- `./scripts/server`
- visit the app on 6543 and register as a new user
- log in
- click on "Contribute" and verify that you are taken to the contribute form
- verify that you are able to upload new lists in the following series of conditions:
    - name and description empty
    - name or description empty
    - name and description but no replaces
    - name, description, and replaces
- verify that a list that has been replaced is listed as "(inactive)" in the dropdown
- click the link to view your lists
- verify that you see the table above
- sign out
- visit :6543/lists and verify that you have an indication that you should sign in
- visit :6543/contribute and verify that you have an indication that you should sign in

kellyi commented on src/app/src/components/FacilityLists.jsx:
```diff
@@ -0,0 +1,122 @@
+                    </Grid>
+                    <Grid
+                        container
+                        classNAme="margin-bottom-64"
```
Typo

kellyi commented:
Looking at the gif above I realized that the replace list select dropdown wasn't very harmonious with the other existing styles, to I force pushed an amended last commit to change its style to look similar to the other inputs:

![screen shot 2019-01-28 at 5 10 58 pm](https://user-images.githubusercontent.com/4165523/51869814-b2aab100-231f-11e9-91dd-abedad20282a.png)

jwalgran commented on src/app/src/actions/upload.js:
```diff
@@ -0,0 +1,49 @@
+
+        window.console.dir(name, description, file);
+
+        // Validate that name does not have any special char:
```
Is this a TODO?

kellyi replied:
I ultimately removed this altogether. The legacy implementation had the validation done client-side in the React component, but I pushed it further out and ultimately we decided that we didn't need these validations yet and could add them server-side if we ever did.

jwalgran commented on src/django/api/views.py:
```diff
@@ -216,14 +215,18 @@ def create(self, request):
 
-        # TODO: Get the organization from the authenticated user
-        organization = Organization.objects.first()
+        organization = Organization.objects.get(name=request.user.name)
```
I ran into a not found exception on this line. The `request.user` object has an `organization` property that should be used here. It's optional, so it needs to be checked for non-None. (Fixing this would take care of #105)

kellyi replied:
Good catch! I'll add this check.

kellyi replied:
Added in 
15c572b along with a test.

jwalgran commented on src/django/api/views.py:
```diff
@@ -262,3 +266,9 @@ def create(self, request):
         return Response(serializer.data)
+
+    def list(self, request):
+        organization = Organization.objects.get(name=request.user.name)
```
The `request.user` object has an `organization` property that should be used here. It's optional, so it needs to be checked for non-None. (Fixing this would take care of #105)

kellyi replied:
Added in 
15c572b along with a test.

kellyi commented:
@jwalgran I added 15c572b to handle validating the user organization as described in #105. I also added two tests to verify that the create and lists method reject with status 400 when the user does not have an organization.

jwalgran commented:
Looking at this now

kellyi commented:
Thanks! Going to merge.
